src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php

```diff
@@ -113,13 +110,16 @@ protected function loadBlocks(array $config, ContainerBuilder $container)
         foreach ($blocks as $blockName => $block) {
             $container
                 ->register(
-                    'elcodi.sitemap_entity_loader.' . $blockName,
-                    '%elcodi.core.sitemap.loader.entity_loader.class%'
+                    'elcodi.sitemap_element_generator.entity_' . $blockName,
+                    'Elcodi\Component\Sitemap\Element\EntitySitemapElementGenerator'
                 )
+                ->addArgument(new Reference('elcodi.factory.sitemap_element'))
```

Comment:
I'd group all of this in a single `setArguments` call.
Maybe is easier to create a new `Definition`, which allows for arguments in its constructor, and then call `$container->addDefinition($definition)`?